### ISSUE TITLE
Special Plants Not Crafting

### ISSUE THREAD

**Issue by faisalsayed10** [hackagotchi/hackagotchi#123]
Special plants aren't able to craft anything somehow but the powders are being applied to them and they are also growing in xp.
[hackstead-bug.mp4.zip](https://github.com/hackagotchi/hackagotchi/files/4937744/hackstead-bug.mp4.zip)

**Comment by Muirrum** [comment]
Define special plants. What are you trying to craft?

**Comment by hkatzdev** [comment]
Its aloe avanta - should be in the vid but not 100% sure.

**Comment by Muirrum** [comment]
In the video I see it happening with a tinkerstool and an aloe avanta.

**Comment by faisalsayed10** [comment]
yeah both the plants

**Comment by hkatzdev** [comment]
I believe this is exclude to @faisalsayed10, everyone else's tinkerstool and aloe avanta work to my knowledge.

**Comment by hkatzdev** [comment]
This might have to do with another bug the another day - it would not show Fayd as online for some reason. I had to run `/hackstead @Fayd` for the server to put him online.

**Comment by lukec11** [comment]
The server was down for a potion of this morning for a migration, @hkatzdev your bug specifically could've been a symptom.

**Comment by faisalsayed10** [comment]
Update: My Tinker and Aloe Still doesn't craft.

**Comment by hkatzdev** [comment]
Huge update - this is an edge case that only Fayd is experiencing due to his unique method. **When not all land plots are filled, crafting breaks.** I have confirmed this by attempting to craft with my tinkerstool twice, both 1min 30sec a part to assure this is not lag. **Removing question label** now that we know the cause.

In order to fix his hackstead temporarily **I have given him 2 bread seeds**. These plants can be destroyed once a fix is in place. Additionally, since his hackstead has been unable to craft for approx 3 days now, **I have also given him 1 rift** to make up for this. I am not a special user so would appreciate reimbursement if you agree that this is fair. 

Since this is a rather large issue I am additionally **pinning this issue** to ensure it will be worked on.

**Comment by faisalsayed10** [comment]
Yeah! Now when I hv plants planted in the other 2 plots, the crafting is working perfect. it is maybe an issue in the code where the crafting is set to null if there are no neighbour plants.

**Comment by faisalsayed10** [comment]
Hi, are there any updates on this?

**Comment by faisalsayed10** [comment]
An Important Update On This Issue:
I tried testing smth and found that there's no need for filling all the plots, but there shud be a neighbour plant. it is regarding the neighbour plant. i just planted a bractus near my tinkerstool but not near the avanta, so now the tinker works properly but avanta doesn't. Conclusion: The bug is regarding neighbour plants. 
Hope this may seem helpful to you!
